src/azul/terra.py

```diff
@@ -272,8 +286,25 @@ def run_sql(self, query: str) -> BigQueryRows:
                     sleep(delay)
                 else:
                     raise e
+            else:
+                if log.isEnabledFor(logging.DEBUG):
+                    log.debug('Job info: %s', json.dumps(self._job_info(job)))
+                return result
         assert False
 
+    def _job_info(self, job: QueryJob) -> JSON:
+        info = job._properties
+        assert isinstance(info, dict)
+        info = {
+            k: v
+            for k, v in info['statistics']['query'].items()
+            if (k != 'referencedTables'
+                and k != 'statementType'
+                and (config.debug >= 2 or k != 'queryPlan'))
+        }
+        info['query'] = self._trunc_query(job.query)
```

Comment:
this could potentially override an existing entry. Probably better to nest than to merge.